src/unix/poll.c

```diff
@@ -32,9 +32,8 @@ static void uv__poll_io(uv_loop_t* loop, uv__io_t* w, unsigned int events) {
   int pevents;
 
   handle = container_of(w, uv_poll_t, io_watcher);
-
-  if (events & POLLERR) {
-    uv__io_stop(loop, w, POLLIN | POLLOUT | UV__POLLRDHUP);
+  if ((events & POLLERR) && (events & UV__POLLPRI) != UV__POLLPRI) {
```

Comment:
Great, that's what we needed to know!  We should probably ifdef the check for linux and add a comment in the code mentioning this so we don't forget.